tools/src/main/java/org/apache/kafka/tools/VerifiableShareConsumer.java

```diff
@@ -0,0 +1,630 @@
+        }
+
+        @JsonProperty
+        public String offsetResetStrategy() {
+            return offsetResetStrategy;
+        }
+    }
+
+    protected static class ShutdownComplete extends ShareConsumerEvent {
+
+        @Override
+        public String name() {
+            return "shutdown_complete";
+        }
+    }
+
+    @JsonPropertyOrder({ "timestamp", "name", "count", "partitions" })
+    public static class RecordsConsumed extends ShareConsumerEvent {
+        private final long count;
+        private final List<RecordSetSummary> partitionSummaries;
+
+        public RecordsConsumed(long count, List<RecordSetSummary> partitionSummaries) {
+            this.count = count;
+            this.partitionSummaries = partitionSummaries;
+        }
+
+        @Override
+        public String name() {
+            return "records_consumed";
+        }
+
+        @JsonProperty
+        public long count() {
+            return count;
+        }
+
+        @JsonProperty
+        public List<RecordSetSummary> partitions() {
+            return partitionSummaries;
+        }
+    }
+
+    @JsonPropertyOrder({ "timestamp", "name", "count", "partitions", "success", "error" })
+    protected static class OffsetsAcknowledged extends ShareConsumerEvent {
+
+        private final long count;
+        private final List<AcknowledgedData> partitions;
+        private final String error;
+        private final boolean success;
+
+        public OffsetsAcknowledged(long count, List<AcknowledgedData> partitions, String error, boolean success) {
+            this.count = count;
+            this.partitions = partitions;
+            this.error = error;
+            this.success = success;
+        }
+
+        @Override
+        public String name() {
+            return "offsets_acknowledged";
+        }
+
+        @JsonProperty
+        public long count() {
+            return count;
+        }
+
+        @JsonProperty
+        public List<AcknowledgedData> partitions() {
+            return partitions;
+        }
+
+        @JsonProperty
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+        public String error() {
+            return error;
+        }
+
+        @JsonProperty
+        public boolean success() {
+            return success;
+        }
+
+    }
+
+    @JsonPropertyOrder({ "timestamp", "name", "key", "value", "topic", "partition", "offset" })
+    public static class RecordData extends ShareConsumerEvent {
+
+        private final ConsumerRecord<String, String> record;
+
+        public RecordData(ConsumerRecord<String, String> record) {
+            this.record = record;
+        }
+
+        @Override
+        public String name() {
+            return "record_data";
+        }
+
+        @JsonProperty
+        public String topic() {
+            return record.topic();
+        }
+
+        @JsonProperty
+        public int partition() {
+            return record.partition();
+        }
+
+        @JsonProperty
```

Comment:
Key and value are both marked as JSON properties, but they're not in the JSON property order. This doesn't seem intentional. Please resolve.

Reply:
My mistake.